src/akushon/config/grpc/config.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2023 Ichiro ITS
+//
+// Permission is hereby granted, free of charge, to any person Zobtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include "akushon/config/grpc/config.hpp"
+
+#include <chrono>
+#include <csignal>
+#include <future>
+#include <string>
+
+#include "akushon/config/utils/config.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "akushon/config/grpc/call_data_get_config.hpp"
+#include "akushon/config/grpc/call_data_save_config.hpp"
+#include "akushon/config/grpc/call_data_publish_set_joints.hpp"
+#include "akushon/config/grpc/call_data_publish_set_torques.hpp"
+#include "akushon/config/grpc/call_data_subscribe_current_joints.hpp"
+#include "akushon/config/grpc/call_data_run_action.hpp"
+
+using grpc::ServerBuilder;
+using namespace std::chrono_literals;
+
+namespace akushon
+{
+ConfigGrpc::ConfigGrpc() {}
+ConfigGrpc::ConfigGrpc(const std::string & path) : path(path) {}
+
+ConfigGrpc::~ConfigGrpc()
+{
+  server_->Shutdown();
+  cq_->Shutdown();
+}
+
+void ConfigGrpc::SignIntHandler(int signum)
+{
+  server_->Shutdown();
+  cq_->Shutdown();
+  exit(signum);
+}
+
+void ConfigGrpc::Run(uint16_t port, const std::string path, rclcpp::Node::SharedPtr node)
+{
+  std::string server_address = absl::StrFormat("0.0.0.0:%d", port);
+
+  ServerBuilder builder;
+  builder.AddListeningPort(server_address, grpc::InsecureServerCredentials());
+  builder.RegisterService(&service_);
+
+  cq_ = builder.AddCompletionQueue();
+  server_ = builder.BuildAndStart();
+  std::cout << "Server listening on " << server_address << std::endl;
+
+  signal(SIGINT, SignIntHandler);
```

Comment:
- Use `std::signal`.
- Use lambda instead of introducing a member function.

```suggestion
  std::signal(SIGINT, [](int signum) {
    server_->Shutdown();
    cq_->Shutdown();
    exit(signum);
  });
```

You probably may need to capture the `server_` and the `cg_` variables properly.